test/e2e/volume-provisioner/framework/framework.go

```diff
@@ -315,3 +302,373 @@ func (f *Framework) newConfigurationProvider(cfg *core.Config) (common.Configura
 	}
 	return conf, nil
 }
+
+// NewClientSetFromFlags builds a kubernetes client from flags.
+func NewClientSetFromFlags() (clientset.Interface, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", TestContext.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	cs, err := clientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return cs, nil
+}
+
+// InstallVolumeProvisioner installs both block and fss volume provisioners and
+// waits for them to be ready.
+func InstallVolumeProvisioner(client clientset.Interface) error {
+	blockDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "oci-block-volume-provisioner",
+			Namespace: "kube-system",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: utilpointer.Int32Ptr(int32(1)),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "oci-block-volume-provisioner"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "oci-block-volume-provisioner"},
+				},
+				Spec: v1.PodSpec{
+					ServiceAccountName: "oci-volume-provisioner",
+					Containers: []v1.Container{{
+						Name:            "oci-block-volume-provisioner",
+						Image:           TestContext.Image,
+						Command:         []string{"/usr/local/bin/oci-volume-provisioner"},
+						ImagePullPolicy: v1.PullAlways,
+						Env: []v1.EnvVar{{
+							Name: "NODE_NAME",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "spec.nodeName",
+								},
+							},
+						}, {
+							Name:  "PROVISIONER_TYPE",
+							Value: core.ProvisionerNameDefault,
+						}},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "config",
+							MountPath: "/etc/oci/",
+							ReadOnly:  true,
+						}},
+					},
+					},
+					Volumes: []v1.Volume{{
+						Name: "config",
+						VolumeSource: v1.VolumeSource{
+							Secret: &v1.SecretVolumeSource{
+								SecretName: "oci-volume-provisioner",
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	// TODO(apryde): Decide whether we're adding --controllers="block,filesystem"
+	// to run both provisioners from the same binary and if not dedup this
+	// code. Otherwise it won't be needed :D
```

Comment:
I wouldn't add this comment here, we can add it at the top of the volume provisioner where the actual change should happen if we decide to do so.